consumer-test/consumerTestController.groovy

```diff
@@ -0,0 +1,213 @@
+TestRunnerThread.libraryVersionUnderTest = LIBRARY_VERSION_UNDER_TEST
+TestRunnerThread.repositoryUnderTest = System.getenv('TRAVIS_REPO_SLUG') ?: 'SAP/jenkins-library'
+
+def testCaseThreads
+def cli = new CliBuilder(
+    usage: 'groovy consumerTestController.groovy [<options>]',
+    header: 'Options:',
+    footer: 'If no options are set all tests are run centrally.')
+
+cli.with {
+    h longOpt: 'help', 'Print this help text and exit.'
+    l longOpt: 'run-locally', 'Run consumer tests locally.'
+    s(longOpt: 'single-test', args: 1, argName: 'filePath', 'Run single test.')
+}
+
+def options = cli.parse(args)
+
+if (options.h) {
+    cli.usage()
+    System.exit 0
+}
+
+if (options.l) {
+    RUNNING_LOCALLY = true
+}
+
+if (!RUNNING_LOCALLY) {
+    if (changeDoesNotNeedConsumerTesting()) {
+        notifyGithub("success", "No consumer tests necessary.")
+        println 'No consumer tests necessary.'
+        System.exit(0)
+    }
+
+    /*
+    In case the build is performed for a pull request TRAVIS_COMMIT is a merge
+    commit between the base branch and the PR branch HEAD. That commit is actually built.
+    But for notifying about a build status we need the commit which is currently
+    the HEAD of the PR branch.
+
+    In case the build is performed for a simple branch (not associated with a PR)
+    In this case there is no merge commit between any base branch and HEAD of a PR branch.
+    The commit which we need for notifying about a build status is in this case simply
+    TRAVIS_COMMIT itself.
+    */
+    COMMIT_HASH = System.getenv('TRAVIS_PULL_REQUEST_SHA') ?: System.getenv('TRAVIS_COMMIT')
+
+    notifyGithub("pending", "Consumer tests are in progress.")
+}
+
+if (options.s) {
+    testCaseThreads = [new TestRunnerThread(options.s)]
+} else {
+    testCaseThreads = listTestCaseThreads()
+}
+
+testCaseThreads.each { it ->
+    it.start()
+}
+
+//The thread below will print to console while the test cases are running.
+//Otherwise the job would be canceled after 10 minutes without output.
+def done = false
+Thread.start {
+    def singleTestCase = testCaseThreads.size() == 1
+    for (; ;) {
+        if (singleTestCase) {
+            testCaseThreads[0].printRunningStdOut()
+        } else {
+            println "[INFO] Consumer tests are still running."
+        }
+
+        if (!singleTestCase && PRINT_LOGS_AFTER_45_MINUTES_COUNTDOWN-- == 0) {
+            testCaseThreads.each { thread ->
+                thread.printOutput()
+            }
+        }
+
+        sleep(AUXILIARY_SLEEP_MS)
+        if (done) {
+            break
+        }
+    }
+}
+
+testCaseThreads.each { it ->
+    it.join()
+}
+done = true
+
+def failedThreads = testCaseThreads.findAll { thread ->
+    thread.returnCode != 0
+}
+
+def status
+def statusMessage
+if (failedThreads.size() == 0) {
+    status = "success"
+    statusMessage = "All consumer tests finished successfully. Congratulations!"
```

Comment:
[meta][friday-afternoon] I would say also a failed test should be awared with "Congratulation". Reason: we found a bug early. We found it before reaching the master branch.

Another example from avionics: Once upon time there as an approach to Salzburg airport during strong crosswinds. The pilots decided for a go-around in the last second. Basically this was not a missed approach, but a sucessful go-around. See [here](http://avherald.com/h?article=4b054ccf)